### PR TITLE
Make sure GH release notes are used correctly with previous tag

### PR DESCRIPTION
**Pull request by AvdLee**
While generating release notes using GH worked as expected, we did not correctly honor `previous_tag_name`. Unfortunately, the create release GH API endpoint does not support this parameter, which is why we have to manually generate the release notes using the [generate release notes API endpoint](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#generate-release-notes-content-for-a-release).

### CODE REVIEW

**Comment by wetransferplatform** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Can't find atom for N_GSYM stabs ___profc_/Users/vagrant/git/Sources/GitBuddy/main.swift:GitBuddy_main in /Users/vagrant/git/build/derived_data/Build/Products/Debug/ExecutableModules/GitBuddy.o
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Can't find atom for N_GSYM stabs ___profd_/Users/vagrant/git/Sources/GitBuddy/main.swift:GitBuddy_main in /Users/vagrant/git/build/derived_data/Build/Products/Debug/ExecutableModules/GitBuddy.o
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>GitBuddyTests: Executed 58 tests (0 failed, 0 retried, 0 skipped) in 0.807 seconds</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  View more details on <a href="https://app.bitrise.io/build/214a69aa-4d09-4c02-b068-2740dbf84f49" target="_blank">Bitrise</a>
  </td>
    </tr>
  </tbody>
</table>

## Code Coverage Report
| Name | Coverage ||
| --- | --- | --- |


### SwiftLint found issues

| Severity | File | Reason |
| -------- | ---- | ------ |
Warning | ReleaseNotesJSON.swift:10 | The disabled 'line_length' rule should be re-enabled before the end of the file (`blanket_disable_command`) |
Warning | ReleaseNotesJSON.swift:10 | SwiftLint rule 'line_length' did not trigger a violation in the disabled region; remove the disable command (`superfluous_disable_command`) |
Warning | ReleaseCommand.swift:36 | Variable declarations should be separated from other statements by a blank line (`let_var_whitespace`) |
Warning | ReleaseCommand.swift:47 | Variable declarations should be separated from other statements by a blank line (`let_var_whitespace`) |
Warning | ReleaseCommand.swift:66 | Variable declarations should be separated from other statements by a blank line (`let_var_whitespace`) |
Error | ReleaseCommand.swift:75 | Line should be 160 characters or less; currently it has 224 characters (`line_length`) |
Warning | ReleaseCommand.swift:21 | Variables should not have redundant type annotation (`redundant_type_annotation`) |
Warning | ReleaseCommand.swift:24 | Variables should not have redundant type annotation (`redundant_type_annotation`) |
Warning | ReleaseCommand.swift:73 | Variables should not have redundant type annotation (`redundant_type_annotation`) |
Warning | ReleaseCommand.swift:76 | Variables should not have redundant type annotation (`redundant_type_annotation`) |
Warning | ReleaseCommand.swift:79 | Variables should not have redundant type annotation (`redundant_type_annotation`) |
Warning | ReleaseCommand.swift:82 | Variables should not have redundant type annotation (`redundant_type_annotation`) |
Warning | Changelog.swift:24 | Colons should be next to the identifier when specifying a type and next to the key in dictionary literals (`colon`) |
Warning | ReleaseProducer.swift:213 | Line should be 140 characters or less; currently it has 142 characters (`line_length`) |
Warning | ReleaseProducer.swift:51 | Use shorthand syntax for optional binding (`shorthand_optional_binding`) |
Warning | ReleaseProducer.swift:111 | Use shorthand syntax for optional binding (`shorthand_optional_binding`) |
Warning | ReleaseProducer.swift:128 | Use shorthand syntax for optional binding (`shorthand_optional_binding`) |
Warning | ReleaseProducer.swift:164 | Use shorthand syntax for optional binding (`shorthand_optional_binding`) |
Warning | Mocks.swift:8 | The disabled 'final_class' rule should be re-enabled before the end of the file (`blanket_disable_command`) |
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against a78bfe1abb90318d0a5702fd0c3963faca270a2c
</p>

File: Sources/GitBuddyCore/GitHub/GitHubReleaseNotesGenerator.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
+//
+//  GitHubReleaseNotesGenerator.swift
+//  
+//
+//  Created by Antoine van der Lee on 13/06/2023.
+//
```
Comment (peagasilva) [inline Sources/GitBuddyCore/GitHub/GitHubReleaseNotesGenerator.swift:6]: Is the project properly configured?

**Review by peagasilva — APPROVED** [review]

File: Sources/GitBuddyCore/Changelog/ChangelogProducer.swift [hunk 2]
```diff
@@ -38,8 +38,20 @@ final class ChangelogProducer: URLSessionInjectable {
...
+    let previousTagName: String?
 
-    init(since: Since = .latestTag, to: Date = Date(), baseBranch: Branch?) throws {
+    init(
+        since: Since = .latestTag,
+        to: Date = Date(),
+        baseBranch: Branch?,
+        useGitHubReleaseNotes: Bool = false,
+        tagName: String? = nil,
+        targetCommitish: String? = nil,
```
Comment (BasThomas) [inline Sources/GitBuddyCore/Changelog/ChangelogProducer.swift:52]: What is a `commitish` referring to here? Can we add a bit of documentation as to what kind of values are expected here?

File: Tests/GitBuddyTests/TestHelpers/ReleaseNotesJSON.swift [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+//  ReleaseNotesJSON.swift
+//  
+//
+//  Created by Antoine van der Lee on 13/06/2023.
+//
+
+import Foundation
+
+// swiftlint:disable line_length
+let ReleaseNotesJSON = """
```
Comment (BasThomas) [inline Tests/GitBuddyTests/TestHelpers/ReleaseNotesJSON.swift:11]: Was a bit surprised to see this uppercased.
```suggestion
let releaseNotesJSON = """
```

File: Tests/GitBuddyTests/TestHelpers/ReleaseNotesJSON.swift [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+{
+  "name": "Release v1.0.0 is now available!",
+  "body": "##Changes in Release v1.0.0 ... ##Contributors @monalisa"
+}
+"""
```
Comment (BasThomas) [inline Tests/GitBuddyTests/TestHelpers/ReleaseNotesJSON.swift:17]: We could also expose this as a `UTF8View` like so
```suggestion
""".utf8
```

Comment (BasThomas) [reply Tests/GitBuddyTests/TestHelpers/ReleaseNotesJSON.swift:17]: ... and then call `Data(releasesNotesJSON)` without needing to unwrap it

**Review by BasThomas — APPROVED** [review]

File: Sources/GitBuddyCore/Changelog/ChangelogProducer.swift (hunk 2 above)
Comment (AvdLee) [reply Sources/GitBuddyCore/Changelog/ChangelogProducer.swift:52]: These parameters are all duplicates of the ChangelogCommand, but it doesn't hurt to duplicate the documentation as well 👌

File: Tests/GitBuddyTests/TestHelpers/ReleaseNotesJSON.swift (hunk 4 above)
Comment (AvdLee) [reply Tests/GitBuddyTests/TestHelpers/ReleaseNotesJSON.swift:17]: Good idea! I took it a little different, but using your suggestion I removed the force unwraps 👌

**Pull request merged**